pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/BrokerCacheUpdaterPeriodic.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.client;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Maintains broker cache this is updated periodically
+ */
+public class BrokerCacheUpdaterPeriodic implements UpdatableBrokerCache {
+  private final BrokerCache _brokerCache;
+  private final ScheduledExecutorService _scheduledExecutorService;
+  private final long _brokerUpdateFreqInMillis;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BrokerCacheUpdaterPeriodic.class);
+
+  public BrokerCacheUpdaterPeriodic(String scheme, String controllerHost,
+      int controllerPort, long brokerUpdateFreqInMillis) {
+    _brokerCache = new BrokerCache(scheme, controllerHost, controllerPort);
+    _scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+    _brokerUpdateFreqInMillis = brokerUpdateFreqInMillis;
+  }
+
+  public void init() throws Exception {
+    try {
+      _brokerCache.updateBrokerData();
+    } catch (Exception e) {
+      LOGGER.error("Broker cache update failed", e);
+      throw e;
```

Comment:
Remove `throw e`

Reply:
updateBrokerData could fail. In that case, we'd want to fail init()

Reply:
then why catch the exception? Doesn't make sense to do that only for logging a message. You can wrap it up in RuntimeException if you want.

Reply:
Ack

Reply:
Ack